test/functionalTest/cases/3727_avoid_triggering_notification_if_metadata_changes_but_not_the_attribute_value_itself/avoid_triggering_notification_if_metadata_changes_but_not_the_attribute_value_itself.test

```diff
@@ -0,0 +1,506 @@
+# 10. Dump the accumulator, see the notification when metadata changes.
+# 11. PATCH /v2/entities/Room2/attrs, to update the value of attribute.
+# 12. Dump the accumulator, see the notification when value of attribute changes.
+# 13. POST /v2/subscriptions, to create subscription with notifyOnMetadataChange=foo and see error.
+#
+
+
+echo '01. POST /v2/entities, to create Room1 with attribute and metadata.'
+echo '==================================================================='
+payload='{
+  "id": "Room1",
+  "type": "Room",
+  "temperature": {
+    "value": 26.5,
+    "type": "Float",
+    "metadata": {
+      "accuracy": {
+        "value": 0.8,
+        "type": "Float"
+      }
+    }
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload"
+echo
+echo
+
+
+
+echo '02. POST /v2/subscriptions, to create subscription with notifyOnMetadataChange=false.'
+echo '====================================================================================='
+payload='{
+  "description": "A subscription to get info about Room1",
+  "subject": {
+    "entities": [
+      {
+        "id": "Room1",
+        "type": "Room"
+      }
+    ],
+    "condition": {
+      "attrs": [
+        "temperature"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "attrs": [
+      "temperature"
+    ],
+    "notifyOnMetadataChange": false
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+subId=$(echo "$_responseHeaders" | grep Location | awk -F/ '{ print $4 }' | tr -d "\r\n")
+echo
+echo
+
+
+
+echo '03. PATCH /v2/entities/Room1/attrs, to update the value of metadata.'
+echo '===================================================================='
+payload='{
+  "temperature": {
+    "value": 26.5,
+    "type": "Float",
+    "metadata": {
+      "accuracy": {
+        "value": 1.8,
+        "type": "Float"
+      }
+    }
+  }
+}'
+orionCurl --url /v2/entities/Room1/attrs --payload "$payload" -X PATCH
+echo
+echo
+
+
+
+echo "04. Dump the accumulator, see no notification when metadata changes."
+echo "===================================================================="
+accumulatorDump
+accumulatorReset
+echo
+echo
+
+
+
+echo '05. PATCH /v2/entities/Room1/attrs, to update the value of attribute.'
+echo '====================================================================='
+payload='{
+  "temperature": {
+    "value": 16.5,
+    "type": "Float",
+    "metadata": {
+      "accuracy": {
+        "value": 1.8,
+        "type": "Float"
+      }
+    }
+  }
+}'
+orionCurl --url /v2/entities/Room1/attrs --payload "$payload" -X PATCH
+echo
+echo
+
+
+
+echo "06. Dump the accumulator, see the notification when value of attribute changes."
+echo "==============================================================================="
+accumulatorDump
+accumulatorReset
+echo
+echo
+
+
+
+echo '07. POST /v2/entities, to create Room1 with temperature and pressure.'
```

Comment:
Maybe the message is wrong... the entity doesn't have pressure attribute

Reply:
Fixed in https://github.com/telefonicaid/fiware-orion/pull/4090/commits/9fbcd9cec499f0dbadee3a2a7a8985f2fd3e9c2b